db/revtree.go

```diff
@@ -128,6 +140,44 @@ func (tree RevTree) UnmarshalJSON(inputjson []byte) (err error) {
 	return
 }
 
+// Validate the revtree.  If it's valid, it will return nil.
+// If invalid, it will return an error with details on why it's invalid.
+func (tree RevTree) Validate() (err error) {
```

Comment:
Can we indicate that this is intended to be test-only in some way?  (maybe ValidateForTest or something) Just so we don't start calling it accidentally in future.